src/CMakeLists.txt

```diff
@@ -1,46 +1,57 @@
-        ${PROJECT_SOURCE_DIR}/include/Simulator.hpp
-        ${CMAKE_CURRENT_SOURCE_DIR}/Simulator.cpp
-        ${PROJECT_SOURCE_DIR}/include/CircuitSimulator.hpp
-        ${CMAKE_CURRENT_SOURCE_DIR}/CircuitSimulator.cpp
-        ${PROJECT_SOURCE_DIR}/include/GroverSimulator.hpp
-        ${CMAKE_CURRENT_SOURCE_DIR}/GroverSimulator.cpp
-        ${PROJECT_SOURCE_DIR}/include/ShorSimulator.hpp
-        ${CMAKE_CURRENT_SOURCE_DIR}/ShorSimulator.cpp
-        ${PROJECT_SOURCE_DIR}/include/ShorFastSimulator.hpp
-        ${CMAKE_CURRENT_SOURCE_DIR}/ShorFastSimulator.cpp
-        ${PROJECT_SOURCE_DIR}/include/StochasticNoiseSimulator.hpp
-        ${CMAKE_CURRENT_SOURCE_DIR}/StochasticNoiseSimulator.cpp
-        ${PROJECT_SOURCE_DIR}/include/DeterministicNoiseSimulator.hpp
-        ${CMAKE_CURRENT_SOURCE_DIR}/DeterministicNoiseSimulator.cpp
-        ${PROJECT_SOURCE_DIR}/include/HybridSchrodingerFeynmanSimulator.hpp
-        ${CMAKE_CURRENT_SOURCE_DIR}/HybridSchrodingerFeynmanSimulator.cpp
-        ${PROJECT_SOURCE_DIR}/include/UnitarySimulator.hpp
-        ${CMAKE_CURRENT_SOURCE_DIR}/UnitarySimulator.cpp
-        ${PROJECT_SOURCE_DIR}/include/PathSimulator.hpp
-        ${CMAKE_CURRENT_SOURCE_DIR}/PathSimulator.cpp
-        )
-target_include_directories(${PROJECT_NAME} PUBLIC $<BUILD_INTERFACE:${${PROJECT_NAME}_SOURCE_DIR}/include>)
+add_library(
+  ${PROJECT_NAME}
+  ${PROJECT_SOURCE_DIR}/include/Simulator.hpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/Simulator.cpp
+  ${PROJECT_SOURCE_DIR}/include/CircuitSimulator.hpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/CircuitSimulator.cpp
+  ${PROJECT_SOURCE_DIR}/include/GroverSimulator.hpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/GroverSimulator.cpp
+  ${PROJECT_SOURCE_DIR}/include/ShorSimulator.hpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/ShorSimulator.cpp
+  ${PROJECT_SOURCE_DIR}/include/ShorFastSimulator.hpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/ShorFastSimulator.cpp
+  ${PROJECT_SOURCE_DIR}/include/StochasticNoiseSimulator.hpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/StochasticNoiseSimulator.cpp
+  ${PROJECT_SOURCE_DIR}/include/DeterministicNoiseSimulator.hpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/DeterministicNoiseSimulator.cpp
+  ${PROJECT_SOURCE_DIR}/include/HybridSchrodingerFeynmanSimulator.hpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/HybridSchrodingerFeynmanSimulator.cpp
+  ${PROJECT_SOURCE_DIR}/include/UnitarySimulator.hpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/UnitarySimulator.cpp
+  ${PROJECT_SOURCE_DIR}/include/PathSimulator.hpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/PathSimulator.cpp)
+target_include_directories(
+  ${PROJECT_NAME}
+  PUBLIC $<BUILD_INTERFACE:${${PROJECT_NAME}_SOURCE_DIR}/include>)
 # set required C++ standard and disable compiler specific extensions
 target_compile_features(${PROJECT_NAME} PUBLIC cxx_std_17)
-set_target_properties(${PROJECT_NAME} PROPERTIES CMAKE_CXX_STANDARD_REQUIRED ON CXX_EXTENSIONS OFF)
+set_target_properties(${PROJECT_NAME} PROPERTIES CMAKE_CXX_STANDARD_REQUIRED ON
+                                                 CXX_EXTENSIONS OFF)
```

Comment:
I am fairly sure that these lines are no longer required as they are covered by the shared project setting from the DDPackage